### ISSUE TITLE
How do I "Use InProcessToolchain to avoid new process creation."?

### ISSUE THREAD

**Issue by AArnott** [dotnet/BenchmarkDotNet#2069]
Benchmarks that used to work just fine are *all* failing with "ExitCode != 0":

```
// **************************
// Benchmark: SerializeBenchmark.JilString: ShortRun(Platform=X64, IterationCount=1, LaunchCount=1, WarmupCount=1)
// *** Execute ***
// Launch: 1 / 1
// Execute: C:\git\MessagePack-CSharp\bin\PerfBenchmarkDotNet\release\net472\d98a1561-6c3f-45ba-a9a4-0cf503a42986.exe --benchmarkName "PerfBenchmarkDotNet.SerializeBenchmark.JilString" --job "ShortRun" --benchmarkId 23 in
// BeforeAnythingElse
ExitCode != 0
No more Benchmark runs will be launched as NO measurements were obtained from the previous run!
```

At the tail end of the log is this warning/tip:

```
// * Warnings *
Environment
  Summary -> Detected error exit code from one of the benchmarks. It might be caused by following antivirus software:
        - Windows Defender (windowsdefender://)
Use InProcessToolchain to avoid new process creation.
```

But no explanation as to how to use this toolchain is provided. I was hoping a command line switch would work but `-?` didn't turn up anything promising.

I'm also curious what would cause this to fail. I know it mentions Defender, but is that a wild guess? Defender doesn't typically get in my way. Why would it here?

**Comment by adamsitnik** [comment]
Hi @AArnott 

> I was hoping a command line switch would work but -? didn't turn up anything promising.

You have missed it:

![image](https://user-images.githubusercontent.com/6011991/182391652-d1770227-e926-4e26-8671-9bd7c76f00f1.png)

We also have some docs with samples: https://benchmarkdotnet.org/articles/configs/toolchains.html#sample-introinprocess

> I'm also curious what would cause this to fail. I know it mentions Defender, but is that a wild guess? Defender doesn't typically get in my way. Why would it here?

Based on the log that you have provided:

> ExitCode != 0

The process has most likely failed with an uncatchable exception as BDN tries to catch and log every exception:

https://github.com/dotnet/BenchmarkDotNet/blob/7fb8723c6e96688cd380b3354397c17d77dfce93/src/BenchmarkDotNet/Templates/BenchmarkProgram.txt#L75-L80

 PTAL at Windows Event Log to see if there is any mention of it.

**Comment by AArnott** [comment]
Thanks for the `-?` help. I had quickly scanned with my eye and then used `findstr` with such args as "toolchain" and "inproc", which apparently didn't match.

The event log does indeed carry details:

```
Application: 5fe118d4-3266-4edf-a4ab-20d38d7e5528.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.NotSupportedException
   at BenchmarkDotNet.Engines.ConsoleHost.SendSignal(BenchmarkDotNet.Engines.HostSignal)
   at BenchmarkDotNet.Autogenerated.UniqueProgramName.AfterAssemblyLoadingAttached(System.String[])
   at BenchmarkDotNet.Autogenerated.UniqueProgramName.Main(System.String[])
```

and
```
Faulting application name: 5fe118d4-3266-4edf-a4ab-20d38d7e5528.exe, version: 0.0.0.0, time stamp: 0xbbdb4721
Faulting module name: KERNELBASE.dll, version: 10.0.22000.795, time stamp: 0x960371d1
Exception code: 0xe0434352
Fault offset: 0x000000000004474c
Faulting process id: 0x9854
Faulting application start time: 0x01d8a676d2a7a7ab
Faulting application path: C:\git\MessagePack-CSharp\bin\PerfBenchmarkDotNet\release\net472\5fe118d4-3266-4edf-a4ab-20d38d7e5528.exe
Faulting module path: C:\WINDOWS\System32\KERNELBASE.dll
Report Id: db969e08-c60f-4fc3-bb93-ef293a81abc0
Faulting package full name: 
Faulting package-relative application ID: 
```

**Comment by adamsitnik** [comment]
@AArnott this particular bug should have been fixed by #1735 and #1940

could you please try our preview version to see if the bug is gone (the fix has not been released to nuget.org yet)?

https://benchmarkdotnet.org/articles/guides/nuget.html#nightly

```xml
<packageSources>
  <add key="bdn-ci" value="https://ci.appveyor.com/nuget/benchmarkdotnet" />
</packageSources>
```

**Comment by AArnott** [comment]
Thank you. Yes, I can confirm that the most recent CI build resolves the issue.
